NitroxModel/DataStructures/NitroxInt3.cs

```diff
@@ -0,0 +1,156 @@
+﻿using System;
+using NitroxModel.DataStructures.GameLogic;
+using ProtoBufNet;
+
+namespace NitroxModel.DataStructures
+{
+    /*
+     * NitroxInt3 allows NitroxModel to be decoupled from Assembly-csharp-firstpass
+     * as it is common code between subnautica and below zero
+     */
+    [Serializable]
+    [ProtoContract]
+    public class NitroxInt3
+    {
+        [ProtoMember(1)]
+        public int X { get; set; }
+
+        [ProtoMember(2)]
+        public int Y { get; set; }
+
+        [ProtoMember(3)]
+        public int Z { get; set; }
+
+        public NitroxInt3()
+        {
+            // For serialization purposes
+        }
+
+        public NitroxInt3(int x, int y, int z)
+        {
+            X = x;
+            Y = y;
+            Z = z;
+        }
+
+        public override string ToString()
+        {
+            return "[Int3 - {" + X + ", " + Y + ", " + Z + "}]";
+        }
+
+        public override bool Equals(object obj)
+        {
+            NitroxInt3 int3 = obj as NitroxInt3;
+
+            return !ReferenceEquals(int3, null) &&
+                   X == int3.X &&
+                   Y == int3.Y &&
+                   Z == int3.Z;
+        }
+
+        public override int GetHashCode()
+        {
+            int hashCode = -307843816;
+            hashCode = hashCode * -1521134295 + X.GetHashCode();
+            hashCode = hashCode * -1521134295 + Y.GetHashCode();
+            hashCode = hashCode * -1521134295 + Z.GetHashCode();
```

Comment:
This will throw with overflow errors under certain configurations if you do not mark it as [`unchecked`](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/unchecked), and will produce an overflow lint failure from SONAR.
```suggestion
            unchecked {
                hashCode = hashCode * -1521134295 + X.GetHashCode();
                hashCode = hashCode * -1521134295 + Y.GetHashCode();
                hashCode = hashCode * -1521134295 + Z.GetHashCode();
            }
```
Additionally, the factors on X, Y, and Z should use different component primes if possible, to avoid hash equivalency for axially orthogonal points.